swift/schema.py

```diff
@@ -811,11 +811,47 @@ class CaseLabelItem(AstNode):
     pattern: Pattern | child
     guard: optional[Expr] | child
 
+class AvailabilitySpec(AstNode):
+    pass
+
+class PlatformVersionAvailabilitySpec(AvailabilitySpec):
+    """
+    An availability spec based on platform and version, for example `macOS 12` or `watchOS 14`
+    """
+    platform: string
+    version: string
+
+class OtherAvailabilitySpec(AvailabilitySpec):
+    """
+    A wildcard availability spec `*`
+    """
+    pass
+
+class AvailabilityInfo(AstNode):
+    """
+    An availability condition of an `if`, `while`, or `guard` statements.
+
+    Examples:
+    ```
+    if #available(iOS 12, *) {
+      // Runs on iOS 12 and above
+    } else {
+      // Runs only anything below iOS 12
+    }
+    if #unavailable(macOS 10.14, *) {
+      // Runs only on macOS 10 and below
+    }
+    ```
+    """
+    unavailability: boolean
```

Comment:
Could this maybe be specified as a `: predicate` instead? I think this would read more naturally in QL then.